### PR TITLE
Kim document detail edit tests

### PR DESCRIPTION
**Pull request by kimallen**
## Description

Adds tests for the `DocumentDetailEdit.jsx`

## Setup

`make client_test`

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](tbd) for this change

### CODE REVIEW

File: src/scenes/Office/DocumentViewer/DocumentDetailEdit.jsx [hunk 1]
```diff
@@ -30,19 +30,30 @@ const DocumentDetailEdit = ({ formValues, moveDocSchema }) => {
     <Fragment>
       <div>
         <FormSection name="moveDocument">
-          <SwaggerField fieldName="title" swagger={moveDocSchema} required />
+          <SwaggerField data-cy="document-title-field" fieldName="title" swagger={moveDocSchema} required />
```
Comment (lynzt) [inline src/scenes/Office/DocumentViewer/DocumentDetailEdit.jsx]: curious about the naming of the `data-cy`... 
below we have things like: `...data-cy="vehicle-make"...` and we know this is in a `moveDocument` form... 
Wondering if we want to call this `title`?

File: src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+      expect(title.length).toEqual(1);
+      expect(moveDocumentType.length).toEqual(1);
+      expect(weightTicketSetType.length).toEqual(1);
+      expect(weightTicketSetType.length).toEqual(1);
+      expect(make.length).toEqual(1);
+      expect(model.length).toEqual(1);
+      expect(vehicleNickname.length).toEqual(0);
+      expect(status.length).toEqual(1);
+      expect(notes.length).toEqual(1);
```
Comment (lynzt) [inline src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx]: I think we could use `contains`: https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/contains.html

File: src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      expect(title.length).toEqual(1);
+      expect(moveDocumentType.length).toEqual(1);
+      expect(weightTicketSetType.length).toEqual(1);
+      expect(make.length).toEqual(0);
+      expect(model.length).toEqual(0);
+      expect(vehicleNickname.length).toEqual(1);
+      expect(emptyWeight.length).toEqual(1);
+      expect(fullWeight.length).toEqual(1);
+      expect(status.length).toEqual(1);
+      expect(notes.length).toEqual(1);
```
Comment (lynzt) [inline src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx]: `contains` here too

File: src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      expect(title.length).toEqual(1);
+      expect(moveDocumentType.length).toEqual(1);
+      expect(weightTicketSetType.length).toEqual(1);
+      expect(make.length).toEqual(0);
+      expect(model.length).toEqual(0);
+      expect(progearType.length).toEqual(1);
+      expect(emptyWeight.length).toEqual(1);
+      expect(fullWeight.length).toEqual(1);
+      expect(status.length).toEqual(1);
+      expect(notes.length).toEqual(1);
```
Comment (lynzt) [inline src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx]: `contains` here too

File: src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+      const notes = documentForm.find('[data-cy="notes"]');
+      const expenseDocumentForm = documentForm.find('ExpenseDocumentForm');
+
+      expect(title.length).toEqual(1);
+      expect(moveDocumentType.length).toEqual(1);
+      expect(storageStartDate.length).toEqual(1);
+      expect(storageEndDate.length).toEqual(1);
+      expect(status.length).toEqual(1);
+      expect(notes.length).toEqual(1);
+      expect(expenseDocumentForm.length).toEqual(1);
```
Comment (lynzt) [inline src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx]: `contains` here too

File: src/scenes/Office/DocumentViewer/DocumentDetailEdit.jsx (hunk 1 above)
Comment (kimallen) [reply src/scenes/Office/DocumentViewer/DocumentDetailEdit.jsx]: updated

File: src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx (hunk 2 above)
Comment (kimallen) [reply src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx]: I ended up using `toHaveProperty()`.  Let me know if that is acceptible.

File: src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx [hunk 6]
```diff
@@ -94,16 +94,16 @@ describe('DocumentDetailEdit', () => {
...
-      expect(emptyWeight.length).toEqual(1);
-      expect(fullWeight.length).toEqual(1);
-      expect(status.length).toEqual(1);
-      expect(notes.length).toEqual(1);
+      expect(title.props()).toHaveProperty('fieldName', 'title');
+      expect(moveDocumentType.props()).toHaveProperty('fieldName', 'move_document_type');
+      expect(weightTicketSetType.props()).toHaveProperty('fieldName', 'weight_ticket_set_type');
+      expect(make.length).toBeFalsy();
+      expect(model.length).toBeFalsy();
+      expect(progearType.props()).toHaveProperty('fieldName', 'vehicle_nickname');
```
Comment (lynzt) [inline src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx]: `vehicle_nickname` and not `progear` - this is prob something that should be changed on the form component

Comment (kimallen) [reply src/scenes/Office/DocumentViewer/DocumentDetailEdit.test.jsx]: @lynzt We actually use the `vehicle_nickname` field for the progear type throughout- the model only has a `vehicle_nickname` and the `progear_type` uses that field.  It's just the label that changes.  Do you think we should create an additional field called `progear_type`?  If so, I could create a story for that.

**Pull request merged**